### PR TITLE
Add necessary files and changes for including the hyena dataset

### PR DESCRIPTION
**Pull request by Cangonin**
This PR:
- adds the configuration files used to pretrain and finetune animal2vec hyena model in the `configs/hyena folder`
- adds a file to generate average precision scores from the .h5 predictions file
- Fixes two bugs in the `scripts/prepare_arb_data_for_audio_pretraining.py`
    - using a list of `futures` with all the audio files will submit all the tasks at once retain them for all the files until the end of the execution of the script, causing the memory to grow progressively and trigger an OOM error eventually. Instead, I streamed the tasks using the `map()` method of the `ThreadPoolExecutor`.
    - When the script is run with the `--randomize-file-names` flag on, all the audio files are put in the same folder, which will cause errors when this number becomes too high. Therefore, I added a `--num-subfolders` to separate the audio files into these subfolders. Right now, this option is only active when the `--randomize-file-names`  flag is on
- Adds the option of selecting some individuals only for the validation set in the file  `scripts/animal2vec_manifest.py`

### CODE REVIEW

File: scripts/animal2vec_manifest.py
Comment (Cangonin) [inline scripts/animal2vec_manifest.py:1]: I left my formatter on before saving, which is why the changes seem substantial. I will highlight what I changed

File: scripts/animal2vec_manifest.py [hunk 1]
```diff
@@ -46,25 +49,52 @@ def get_parser():
...
     )
+    parser.add_argument(
+        "--valid-set-individuals",
+        type=str,
+        default="",
+        nargs="+",
+        help="List of individual IDs that should be exclusively used for the validation set for the fine-tuning phase."
+        "This is useful to test generalization to unseen individuals."
+        "Will be ignored when leave-p-out is set.",
+    )
```
Comment (Cangonin) [inline scripts/animal2vec_manifest.py:89]: Added flag

File: scripts/animal2vec_manifest.py [hunk 2]
```diff
@@ -101,6 +131,53 @@ def flatten(l):
...
+        idx_ind_valid = get_files_indices_one_individual(
+            individual, labels_X, id_lookup_file_path
+        )
+        valid_set_indices.extend(idx_ind_valid)
+
+    train_set_indices = list(set(range(len(labels_X))) - set(valid_set_indices))
+    it_ = iter([(np.array(train_set_indices), np.array(valid_set_indices))])
+    return it_
+
+
```
Comment (Cangonin) [inline scripts/animal2vec_manifest.py:180]: Added code

File: scripts/animal2vec_manifest.py [hunk 3]
```diff
@@ -185,61 +276,111 @@ def main(args):
...
+        )
+        it_ = (
+            sss.split(labels_X, targets)
+            if 0 < args.valid_percent
+            else (np.arange(len(labels_X)), [])
+        )
+    else:
+        it_ = split_train_valid_by_individuals(
+            labels_X, args.valid_set_individuals, args.id_lookup_file_path
+        )
```
Comment (Cangonin) [inline scripts/animal2vec_manifest.py:340]: Added code

**Comment by julian-carpenter** [comment]
looks great, thanks you
